lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -676,7 +676,10 @@ def stub_forge(forge_host = nil)
         def sleep_until_puppetdb_started(host, nonssl_port = nil, ssl_port = nil)
           nonssl_port = options[:puppetdb_port_nonssl] if nonssl_port.nil?
           ssl_port = options[:puppetdb_port_ssl] if ssl_port.nil?
-          curl_with_retries("start puppetdb", host, "http://localhost:#{nonssl_port}", 0, 120)
+          endpoint = 'status/v1/services/puppetdb-status'
+          retry_on(host,
+                   "curl -m 1 http://localhost:#{nonssl_port}/#{endpoint} | grep '\"state\":\"running\"'",
+                   {:max_retries => 120})
           curl_with_retries("start puppetdb (ssl)",
```

Comment:
@tvpartytonight The call to `curl_with_retries` validates that the {{ssl}} port is open after we have verified that the service is available on the {{nonssl_port}}. So, I believe it is still required in order to make sure that the {{ssl}} port has been successfully opened.